smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/tracing/HeaderInjectAdapter.java

```diff
@@ -0,0 +1,18 @@
+package io.smallrye.reactive.messaging.kafka.tracing;
+
+import java.nio.charset.StandardCharsets;
+
+import org.apache.kafka.common.header.Headers;
+
+import io.opentelemetry.context.propagation.HttpTextFormat;
+
+public class HeaderInjectAdapter implements HttpTextFormat.Setter<Headers> {
+    public static final HeaderInjectAdapter SETTER = new HeaderInjectAdapter();
+
+    @Override
+    public void set(Headers headers, String key, String value) {
+        if (headers != null) {
+            headers.remove(key).add(key, value.getBytes(StandardCharsets.UTF_8));
```

Comment:
Can't recall right now, but this is the pattern used in the Kafka instrumentation class

Reply:
The equals check on `RecordHeader` uses key and value, so I think it's necessary to remove the previous one using the key otherwise you could end up with different values as the underlying collection is a `List`.

Maybe I'm wrong, but probably better to be safe

Reply:
I was thinking add was appending, but let's check.

Reply:
Yes add is appending, so we need to remove the previous value to ensure we don't end up with multiples present